python/beeai_framework/agents/react/types.py

```diff
@@ -18,17 +19,19 @@
     UserPromptTemplateInput,
 )
 from beeai_framework.agents.types import AgentExecutionConfig, AgentMeta, BaseAgentRunOptions
-from beeai_framework.backend import AssistantMessage
+from beeai_framework.backend import AnyMessage
 from beeai_framework.backend.chat import ChatModel
 from beeai_framework.backend.types import ChatModelOutput
 from beeai_framework.memory.base_memory import BaseMemory
 from beeai_framework.template import PromptTemplate
 from beeai_framework.tools.tool import AnyTool
 from beeai_framework.utils.strings import to_json
+from beeai_framework.utils.warnings import deprecated_type_alias
 
 
 class ReActAgentRunInput(BaseModel):
-    prompt: str | None = None
+    model_config = ConfigDict(arbitrary_types_allowed=True)
```

Comment:
`arbitrary_types_allowed` is not needed, use `prompt: str | list[InstanceOf[AnyMessage]]` instead